### PR TITLE
Render mermaid diagram in markdown

### PR DESCRIPTION
**Pull request by Chrisyunhua**
Add support for mermaid in the markdown format

here is an example: https://paste.chrisyunhua.com/BSgcltfX2I5XkwGi2iuss

### CODE REVIEW

**Comment by mkaczanowski** [comment]
Hi,
this looks great and thanks for the PR!

The intention of this Pastebin impl is to keep things as simple as possible. I imagine mermaid js is not going to be used by the majority, so I would rather gate the feature.

What I propose is to pluginify the current code, to accommodate custom plugins such as mermaid.js.

File: static/index.html [hunk 1]
```diff
@@ -309,6 +309,7 @@ <h5 class="modal-title" id="password-modal-label">
     <script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-notify/0.2.0/js/bootstrap-notify.min.js"></script>
     <script src="{{uri_prefix}}/static/prism.js"></script>
+    <script src="{{uri_prefix}}/static/mermaid.min.js"></script>
```
Comment (mkaczanowski) [inline static/index.html:312]: there is no need to host the .js file:

https://cdnjs.com/libraries/mermaid

File: static/custom.js [hunk 2]
```diff
@@ -52,6 +53,7 @@ $(document).ready(function() {
 		if ($("#pastebin-code-block").length) {
 			$('#pastebin-code-block').attr('class', 'language-' + $("#language-selector").val());
 			Prism.highlightElement($('#pastebin-code-block')[0]);
+			initMermaid();
```
Comment (mkaczanowski) [inline static/custom.js:56]: taken this block is duplicated I would move it to separate function

**Comment by mkaczanowski** [comment]
@Chrisyunhua I'll refactor code a bit and I'll get back to you

**Comment by Chrisyunhua** [comment]
Sorry for the late replay.

To accommodate custom plugins(only **frontend** related), how about adding a parameter `--plugins` to the entrypoint which accepts a list of js url and render them as `<script>` in the `index.html`, then we could run the app with:
```
pastebin --plugins https://url/to/plugin.js
```

**Comment by mkaczanowski** [comment]
@Chrisyunhua I did a small refactoring to accommodate plugins such as mermaid.js, please see:
https://github.com/mkaczanowski/pastebin/commit/a8e8476c9c25575d5f8a29431bff3d4d5459ac0c

Now you can pass:
```
--plugins prism mermaid
```

and that should work :)

**Comment by Chrisyunhua** [comment]
THX, I've tried the latest docker image, it looks really cool!

**Pull request closed without merging**